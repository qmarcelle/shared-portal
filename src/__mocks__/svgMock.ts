import React from 'react';

const SvgMock = React.forwardRef<
  HTMLSpanElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return React.createElement('span', {
    ref,
    ...props,
    'data-testid': 'svg-mock',
  });
});

SvgMock.displayName = 'SvgMock';

export default SvgMock;
export const ReactComponent = SvgMock;
