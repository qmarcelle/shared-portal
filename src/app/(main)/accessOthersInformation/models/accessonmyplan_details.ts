export interface AccessOnMyPlanDetails {
  memberName: string;
  DOB: string;
  isOnline: boolean;
}
