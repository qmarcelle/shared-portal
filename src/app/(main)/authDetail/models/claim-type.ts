export enum ClaimType {
  Medical = 'Medical',
  Pharmacy = 'Pharmacy',
  Dental = 'Dental',
}
