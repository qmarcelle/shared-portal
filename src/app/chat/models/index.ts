// Export all chat-related types
export type {
  CobrowseConfig,
  CobrowseInitResponse,
  CobrowseSession,
  CobrowseSessionResponse,
} from './cobrowse';

// Make sure the ChatMessage type is consistent by defining it inline here
export interface ChatMessage {
  id: string;
  text: string;
  sender: 'user' | 'agent' | 'bot';
  timestamp: Date;
}

export * from './plans';
export type {
  ChatEligibility,
  ChatPayload,
  ChatSession,
  ChatSessionJWT,
  ChatState,
  PlanSwitcherState,
} from './session';
export type {
  BusinessDay,
  BusinessHours,
  ChatConfig,
  ChatOption,
  CobrowseState,
  UserEligibility,
} from './types';
