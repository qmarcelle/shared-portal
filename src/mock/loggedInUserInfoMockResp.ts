export const loggedInUserInfoMockResp = {
  isActive: true,
  subscriberLoggedIn: true,
  lob: 'REGL',
  groupData: {
    groupID: '100000',
    groupCK: '21908',
    groupName: 'Chris B Hall Enterprises',
    parentGroupID: '100001',
    subGroupID: '0001',
    subGroupCK: 28951,
    subGroupName: 'Chris B Hall Enterprises',
    clientID: 'EI',
    policyType: 'INT',
    groupEIN: '620427913',
  },
  networkPrefix: 'QMI',
  subscriberID: '902218823',
  subscriberCK: '91722400',
  subscriberFirstName: 'CHRIS',
  subscriberLastName: 'HALL',
  subscriberTitle: '',
  subscriberDateOfBirth: '08/06/1959',
  subscriberOriginalEffectiveDate: '01/01/2001',
  members: [
    {
      isActive: true,
      memberOrigEffDt: '06/29/2009',
      memberCk: 91722407,
      firstName: 'CHRISTMAS',
      middleInitial: '',
      lastName: 'HALL',
      title: '',
      memRelation: 'S',
      birthDate: '06/29/2009',
      gender: 'M',
      memberSuffix: 6,
      mailAddressType: 'H',
      workPhone: '',
      otherInsurance: [],
      coverageTypes: [
        {
          productType: 'M',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'S',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'V',
          coverageLevel: 'A',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
        {
          productType: 'D',
          coverageLevel: 'A',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
      ],
      planDetails: [
        {
          productCategory: 'M',
          planID: 'MBPX0806',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Blue Network P',
        },
        {
          productCategory: 'S',
          planID: 'WSXM0218',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Wellness Plan',
        },
        {
          productCategory: 'V',
          planID: 'VEMGN002',
          effectiveDate: 1546318800000,
          planStartDate: 1383278400000,
          planClassID: 'PPOA',
          networkPlanName: 'Vision Blue',
        },
        {
          productCategory: 'D',
          planID: 'DEHCNY02',
          effectiveDate: 1546318800000,
          planStartDate: 1383278400000,
          planClassID: 'PPOA',
          networkPlanName: 'Vision Blue',
        },
      ],
      inXPermissions: true,
      futureEffective: false,
      loggedIn: false,
      hasSocial: true,
      esipharmacyEligible: true,
    },
    {
      isActive: true,
      memberOrigEffDt: '06/15/2009',
      memberCk: 91722406,
      firstName: 'KRISSY',
      middleInitial: 'C',
      lastName: 'HALL',
      title: '',
      memRelation: 'D',
      birthDate: '06/15/2009',
      gender: 'F',
      memberSuffix: 5,
      mailAddressType: 'H',
      workPhone: '',
      otherInsurance: [],
      coverageTypes: [
        {
          productType: 'M',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'S',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'V',
          coverageLevel: 'A',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
      ],
      planDetails: [
        {
          productCategory: 'M',
          planID: 'MBPX0806',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Blue Network P',
        },
        {
          productCategory: 'S',
          planID: 'WSXM0218',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Wellness Plan',
        },
        {
          productCategory: 'V',
          planID: 'VEMGN002',
          effectiveDate: 1546318800000,
          planStartDate: 1383278400000,
          planClassID: 'PPOA',
          networkPlanName: 'Vision Blue',
        },
      ],
      inXPermissions: true,
      futureEffective: false,
      loggedIn: false,
      hasSocial: false,
      esipharmacyEligible: true,
    },
    {
      isActive: true,
      memberOrigEffDt: '05/01/2009',
      memberCk: 91722405,
      firstName: 'CHRISTIAN',
      middleInitial: '',
      lastName: 'HALL',
      title: '',
      memRelation: 'S',
      birthDate: '10/31/2011',
      gender: 'M',
      memberSuffix: 4,
      mailAddressType: 'H',
      workPhone: '',
      otherInsurance: [],
      coverageTypes: [
        {
          productType: 'M',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'S',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'V',
          coverageLevel: 'A',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
      ],
      planDetails: [
        {
          productCategory: 'M',
          planID: 'MBPX0806',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Blue Network P',
        },
        {
          productCategory: 'S',
          planID: 'WSXM0218',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Wellness Plan',
        },
        {
          productCategory: 'V',
          planID: 'VEMGN002',
          effectiveDate: 1546318800000,
          planStartDate: 1383278400000,
          planClassID: 'PPOA',
          networkPlanName: 'Vision Blue',
        },
      ],
      inXPermissions: true,
      futureEffective: false,
      loggedIn: false,
      hasSocial: true,
      esipharmacyEligible: true,
    },
    {
      isActive: true,
      memberOrigEffDt: '01/01/2001',
      memberCk: 91722402,
      firstName: 'KRISTY',
      middleInitial: '',
      lastName: 'HALL',
      title: '',
      memRelation: 'W',
      birthDate: '05/17/1971',
      gender: 'F',
      memberSuffix: 1,
      mailAddressType: 'H',
      workPhone: '',
      otherInsurance: [],
      coverageTypes: [
        {
          productType: 'S',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'V',
          coverageLevel: 'A',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
      ],
      planDetails: [
        {
          productCategory: 'S',
          planID: 'WSXM0218',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Wellness Plan',
        },
        {
          productCategory: 'V',
          planID: 'VEMGN002',
          effectiveDate: 1546318800000,
          planStartDate: 1383278400000,
          planClassID: 'PPOA',
          networkPlanName: 'Vision Blue',
        },
      ],
      inXPermissions: true,
      futureEffective: false,
      loggedIn: false,
      hasSocial: true,
      esipharmacyEligible: false,
    },
    {
      isActive: true,
      memberOrigEffDt: '01/01/2001',
      memberCk: 91722401,
      firstName: 'CHRIS',
      middleInitial: 'B',
      lastName: 'HALL',
      title: '',
      memRelation: 'M',
      birthDate: '08/06/1959',
      gender: 'M',
      memberSuffix: 0,
      mailAddressType: 'H',
      workPhone: '1234567890',
      otherInsurance: [],
      coverageTypes: [
        {
          productType: 'D',
          coverageLevel: '*',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
        {
          productType: 'M',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'S',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'V',
          coverageLevel: 'A',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
      ],
      planDetails: [
        {
          productCategory: 'D',
          planID: 'DEHCNY02',
          effectiveDate: 1509508800000,
          planStartDate: 1451624400000,
          planClassID: 'PPOA',
          networkPlanName: 'DentalBlue Preferred Network',
        },
        {
          productCategory: 'M',
          planID: 'MBPX0806',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Blue Network P',
        },
        {
          productCategory: 'S',
          planID: 'WSXM0218',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Wellness Plan',
        },
        {
          productCategory: 'V',
          planID: 'VEMGN002',
          effectiveDate: 1546318800000,
          planStartDate: 1383278400000,
          planClassID: 'PPOA',
          networkPlanName: 'Vision Blue',
        },
      ],
      inXPermissions: true,
      futureEffective: false,
      loggedIn: true,
      hasSocial: true,
      esipharmacyEligible: true,
    },
    {
      isActive: true,
      memberOrigEffDt: '01/01/2014',
      memberCk: 91722409,
      firstName: 'CHRISTOFF',
      middleInitial: '',
      lastName: 'HALL',
      title: '',
      memRelation: 'S',
      birthDate: '01/01/2000',
      gender: 'M',
      memberSuffix: 8,
      mailAddressType: 'H',
      workPhone: '',
      otherInsurance: [],
      coverageTypes: [
        {
          productType: 'M',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'S',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'V',
          coverageLevel: 'A',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
      ],
      planDetails: [
        {
          productCategory: 'M',
          planID: 'MBPX0806',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Blue Network P',
        },
        {
          productCategory: 'S',
          planID: 'WSXM0218',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Wellness Plan',
        },
        {
          productCategory: 'V',
          planID: 'VEMGN002',
          effectiveDate: 1546318800000,
          planStartDate: 1383278400000,
          planClassID: 'PPOA',
          networkPlanName: 'Vision Blue',
        },
      ],
      inXPermissions: true,
      futureEffective: false,
      loggedIn: false,
      hasSocial: false,
      esipharmacyEligible: true,
    },
    {
      isActive: true,
      memberOrigEffDt: '12/11/2006',
      memberCk: 91722408,
      firstName: 'CHRISTO',
      middleInitial: '',
      lastName: 'HALL',
      title: '',
      memRelation: 'S',
      birthDate: '12/11/2006',
      gender: 'M',
      memberSuffix: 7,
      mailAddressType: 'H',
      workPhone: '',
      otherInsurance: [],
      coverageTypes: [
        {
          productType: 'M',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'S',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'V',
          coverageLevel: 'A',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
      ],
      planDetails: [
        {
          productCategory: 'M',
          planID: 'MBPX0806',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Blue Network P',
        },
        {
          productCategory: 'S',
          planID: 'WSXM0218',
          effectiveDate: 1546318800000,
          planStartDate: 1514782800000,
          planClassID: 'PPOA',
          networkPlanName: 'Wellness Plan',
        },
        {
          productCategory: 'V',
          planID: 'VEMGN002',
          effectiveDate: 1546318800000,
          planStartDate: 1383278400000,
          planClassID: 'PPOA',
          networkPlanName: 'Vision Blue',
        },
      ],
      inXPermissions: true,
      futureEffective: false,
      loggedIn: false,
      hasSocial: true,
      esipharmacyEligible: true,
    },
  ],
  coverageTypes: [
    {
      productType: 'M',
      coverageLevel: 'A',
      exchange: false,
      indvGroupInd: '',
      pedAdultInd: '',
    },
    {
      productType: 'S',
      coverageLevel: 'A',
      exchange: false,
      indvGroupInd: '',
      pedAdultInd: '',
    },
    {
      productType: 'V',
      coverageLevel: 'A',
      exchange: true,
      indvGroupInd: 'Group',
      pedAdultInd: 'Adult',
    },
    {
      productType: 'D',
      coverageLevel: '*',
      exchange: true,
      indvGroupInd: 'Group',
      pedAdultInd: 'Adult',
    },
  ],
  addresses: [
    {
      type: '1',
      address1: 'TEST BLUE ACCESS',
      address2: '',
      address3: '',
      city: 'Chattanooga',
      state: 'TN',
      zipcode: '37412',
      county: 'HAMILTON',
      phone: '',
      email: '',
    },
    {
      type: '2',
      address1: 'TEST 2 BLUE ACCESS',
      address2: '',
      address3: '',
      city: 'Chattanooga',
      state: 'TN',
      zipcode: '37402',
      county: 'HAMILTON',
      phone: '',
      email: '',
    },
    {
      type: 'H',
      address1: '1 CAMERON HILL CIRCLE',
      address2: '',
      address3: '',
      city: 'CHATTANOOGA',
      state: 'TN',
      zipcode: '37402',
      county: 'HAMILTON',
      phone: '4235353065',
      email: '',
    },
  ],
  authFunctions: [
    {
      functionName: 'MULTITIER',
      available: true,
    },
    {
      functionName: 'MULTITIER',
      available: true,
    },
    {
      functionName: 'MULTITIER',
      available: true,
    },
    {
      functionName: 'MULTITIER',
      available: true,
    },
    {
      functionName: 'MULTITIER',
      available: true,
    },
    {
      functionName: 'MULTITIER',
      available: true,
    },
    {
      functionName: 'MDLIVE',
      available: false,
    },
    {
      functionName: 'OHDELIGIBLE',
      available: false,
    },
    {
      functionName: 'MedicareAdvantageGroupIndicator',
      available: false,
    },
    {
      functionName: 'AllMedicareAdvantage',
      available: false,
    },
    {
      functionName: 'TELEHEALTH_BEHAVIORAL',
      available: true,
    },
    {
      functionName: 'QUEST_SELECT',
      available: false,
    },
    {
      functionName: 'TELADOC',
      available: false,
    },
    {
      functionName: 'TELADOC_PRIMARY360',
      available: false,
    },
    {
      functionName: 'TELADOC_MYSTRENGTHCOMPLETE',
      available: false,
    },
    {
      functionName: 'TELADOC_DIABETESPREVENTION',
      available: false,
    },
    {
      functionName: 'TELADOC_DIABETESMGMT',
      available: false,
    },
    {
      functionName: 'TELADOC_HYPERTENSIONMGMT',
      available: false,
    },
    {
      functionName: 'MedicalWellnesswithIncentives',
      available: false,
    },
    {
      functionName: 'PHACommercialEligible',
      available: true,
    },
    {
      functionName: 'HealthReimbursementAccount',
      available: false,
    },
    {
      functionName: 'FlexibleSpendingAccount',
      available: false,
    },
    {
      functionName: 'UpdateSSNIneligible',
      available: false,
    },
    {
      functionName: 'PBP_INELIGIBLE',
      available: false,
    },
    {
      functionName: 'PHARMACYINTEGRATEDGROUP',
      available: true,
    },
    {
      functionName: 'CLAIMSHOLD',
      available: false,
    },
    {
      functionName: 'DELINQUENCY',
      available: false,
    },
    {
      functionName: 'COBUPDATEELIGIBLE',
      available: true,
    },
    {
      functionName: 'COBRAELIGIBLE',
      available: true,
    },
    {
      functionName: 'HEALTHCOACHELIGIBLE',
      available: true,
    },
    {
      functionName: 'BLUEHEALTHREWARDS',
      available: false,
    },
    {
      functionName: 'CHIPELIGIBLE',
      available: false,
    },
    {
      functionName: 'PHSELIGIBLE',
      available: true,
    },
    {
      functionName: 'CHAT_ELIGIBLE',
      available: true,
    },
    {
      functionName: 'PCAWELLNESSELIGIBLE',
      available: false,
    },
    {
      functionName: 'HEALTHYMATERNITY',
      available: true,
    },
    {
      functionName: 'LIVONGO',
      available: true,
    },
    {
      functionName: 'LIVONGODMP',
      available: true,
    },
    {
      functionName: 'ONLIFEWELLNESSELIGIBLE',
      available: true,
    },
    {
      functionName: 'BLUEPRKS',
      available: true,
    },
    {
      functionName: 'TEMPIDEXCLUSION',
      available: false,
    },
    {
      functionName: 'MYPCPELIGIBLE',
      available: true,
    },
    {
      functionName: 'IDPROTECTELIGIBLE',
      available: true,
    },
    {
      functionName: 'ENROLLELIGIBLE',
      available: true,
    },
    {
      functionName: 'FSAHRAELIGIBLE',
      available: true,
    },
    {
      functionName: 'DENTALCOSTELIGIBLE',
      available: true,
    },
    {
      functionName: 'ENABLE_COST_TOOLS',
      available: true,
    },
    {
      functionName: 'ENABLE_COST_EST_TOOL',
      available: true,
    },
    {
      functionName: 'INDIVIDUAL_SBC_ELIGIBLE',
      available: false,
    },
    {
      functionName: 'ENABLE_BENEFIT_CHANGE_TAB',
      available: false,
    },
    {
      functionName: 'SOLERAELIGIBLE',
      available: false,
    },
    {
      functionName: 'ENABLE_PHAR_TAB',
      available: true,
    },
    {
      functionName: 'RX_MEDICARE',
      available: false,
    },
    {
      functionName: 'RX_PREFERRED_ELIGIBLE',
      available: true,
    },
    {
      functionName: 'SANITAS_ELIGIBLE',
      available: false,
    },
    {
      functionName: 'CONDENSED_EXPERIENCE',
      available: false,
    },
    {
      functionName: 'HINGE_HEALTH_ELIGIBLE',
      available: true,
    },
    {
      functionName: 'OTCEnable',
      available: false,
    },
    {
      functionName: 'careManagementExclusion',
      available: true,
    },
    {
      functionName: 'COBUPDATEREMINDER',
      available: true,
    },
    {
      functionName: 'EMBOLD_HEALTH',
      available: false,
    },
    {
      functionName: 'GROUP_RENEWAL_DATE_BEFORE_TODAY',
      available: true,
    },
  ],
  healthCareAccounts: [],
  esigroupNum: '100000MBPX0806',
  cmcondition: [],
};
