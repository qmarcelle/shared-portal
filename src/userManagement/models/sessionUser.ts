import { VisibilityRules } from '@/visibilityEngine/rules';

export enum UserRole {
  MEMBER = 'MEM',
  PERSONAL_REP = 'PR',
  AUTHORIZED_USER = 'AU',
  NON_MEM = 'NM',
}

export type SessionUser = {
  id: string;
  currUsr: {
    umpi: string;
    fhirId: string;
    role: UserRole;
    plan:
      | {
          memCk: string;
          sbsbCk: string;
          grgrCk: string;
          grpId: string;
          subId: string;
          fhirId: string;
          ntwkId: string;
        }
      | undefined;
  };
  rules?: string;
  vRules?: VisibilityRules;
  subscriberId?: string;
};
