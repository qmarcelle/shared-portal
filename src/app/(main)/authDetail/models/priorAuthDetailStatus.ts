import { ClaimStatus } from '@/app/(main)/authDetail/models/claim-status';
import { ClaimType } from '@/app/(main)/authDetail/models/claim-type';

export interface PriorAuthDetailsStatus {
  priorAuthDetailType: ClaimType;
  priorAuthDetailName: string;
  dateOfVisit: string;
  priorAuthDetailStatus: ClaimStatus;
  member: string;
  PriorAuthReferenceId: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  authInfo: any;
  referredName: string;
}
