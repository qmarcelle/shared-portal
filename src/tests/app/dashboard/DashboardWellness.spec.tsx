import DashboardPage from '@/app/dashboard/page';
import { loggedInUserInfoMockResp } from '@/mock/loggedInUserInfoMockResp';
import { mockedAxios } from '@/tests/__mocks__/axios';
import { mockedFetch } from '@/tests/setup';
import { fetchRespWrapper } from '@/tests/test_utils';
import { UserRole } from '@/userManagement/models/sessionUser';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = async () => {
  const page = await DashboardPage();
  return render(page);
};

jest.mock('../../../auth', () => ({
  auth: jest.fn(() =>
    Promise.resolve({
      user: {
        currUsr: {
          firstName: 'Chris',
          role: UserRole.MEMBER,
          plan: {
            planName: 'BlueCross BlueShield of Tennessee',
            subId: '123456',
            grpId: '100000',
            memCk: '123456789',
            coverageType: ['Medical', 'Dental', 'Vision'],
          },
        },
        vRules: {
          isWellnessQa: true,
        },
      },
    }),
  ),
}));

describe('Dashboard Page for Wellness Only', () => {
  it('should render UI correctly', async () => {
    mockedFetch.mockResolvedValueOnce(
      fetchRespWrapper(loggedInUserInfoMockResp),
    );
    mockedAxios.get.mockResolvedValueOnce({
      data: {
        physicianId: '3118777',
        physicianName: 'Louthan, James D.',
        address1: '2033 Meadowview Ln Ste 200',
        address2: '',
        address3: '',
        city: 'Kingsport',
        state: 'TN',
        zip: '376607432',
        phone: '4238572260',
        ext: '',
        addressType: '1',
        taxId: '621388079',
      },
    });
    mockedAxios.get.mockResolvedValueOnce({
      data: [
        {
          id: '45',
          providedBy: 'Davis Vision',
          contact: '1-800-456-9876',
          url: 'https://davis-vision.com',
        },
        {
          id: '87',
          providedBy: 'Nirmal Dental',
          contact: '1-800-367-9676',
          url: 'https://nirmaldental.com',
        },
        {
          id: '25',
          providedBy: 'Low Pharm',
          contact: '1-800-834-2465',
        },
        {
          id: '289',
          providedBy: 'Quant Labs',
          contact: '1-800-834-3465',
        },
      ],
    });
    mockedAxios.get.mockResolvedValueOnce({
      data: {
        currentPolicies: [
          {
            memberCk: '502622001',
            subscriberName: 'REBEKAH WILSON',
            groupName: 'Radio Systems Corporation',
            memberId: '90447969100',
            planTypes: ['M', 'R', 'S'],
            amplifyMember: false,
          },
        ],
        pastPolicies: [
          {
            memberCk: '846239401',
            subscriberName: 'JOHNATHAN ANDERL',
            groupName: 'Ruby Tuesday Operations LLC',
            memberId: '90865577900',
            planTypes: ['D', 'R', 'S', 'M', 'V'],
            amplifyMember: false,
          },
        ],
      },
    });
    const component = await renderUI();
    expect(mockedAxios.get).toHaveBeenCalledWith(
      '/api/member/v1/members/byMemberCk/123456789/benefits/employerProvidedBenefits',
    );

    expect(screen.getByText('Member Wellness Center')).toBeVisible();
    expect(screen.getByText('Member Discount')).toBeVisible();
    expect(component.baseElement).toMatchSnapshot();
  });
});
