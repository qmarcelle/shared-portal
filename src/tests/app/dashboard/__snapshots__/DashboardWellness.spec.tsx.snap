// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page for Wellness Only should render UI correctly 1`] = `
<body>
  <div>
    <div
      class="flex flex-col justify-center items-center page"
    >
      <div
        class="flex flex-col h-175px w-full text-white justify-center items-center brand-gradient px-4 "
      >
        <div
          class="flex flex-col items-start w-full app-content px-4"
        >
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <h1
            class="title-1 font-thin"
          >
            Welcome, 
             
            Chris
          </h1>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <p
            class="body-1"
            style="display: block; max-width: 100%;"
          >
            Plan: Chris B Hall Enterprises
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
          />
          <p
            class="body-1"
            style="display: block; max-width: 100%;"
          >
            Subscriber ID: 902218823
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
          />
          <p
            class="body-1"
            style="display: block; max-width: 100%;"
          >
            Group ID: 100000
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
          />
          <p
            class="body-1"
            style="display: block; max-width: 100%;"
          >
            Policies: Medical, Wellness, Vision, Dental
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <a
            class="link-white-text"
            href="/member/myplan"
          >
            <p
              class="pb-2 pt-2"
            >
              View Plan Details
            </p>
          </a>
        </div>
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
      />
      <div
        class="flex flex-col app-content app-base-font-color"
      >
        <section>
          <div
            class="card-main large-section skip-underscore"
          >
            <div
              class="flex flex-col"
            >
              <div
                class="flex flex-row"
              >
                <img
                  alt=""
                  class="size-10"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <section
                  class="flex justify-start self-start my-health-link"
                >
                  <span
                    class="rich-text body-1 undefined"
                  >
                    <div
                      class="flex flex-row body-1 flex-grow md:!flex !block align-top mt-4 ml-2"
                    >
                      <a
                        aria-label="Member Wellness Center"
                        href=""
                        style="display: block;"
                      >
                        <button
                          class="flex flex-row link-container link hover:!underline caremark !flex pt-0 pl-0"
                          style="max-width: max-content; height: auto; display: block;"
                          tabindex="-1"
                          type="button"
                        >
                          <p
                            class="link undefined"
                          >
                            Member Wellness Center
                          </p>
                          <p
                            class="ml-1"
                          >
                            <img
                              alt=""
                              data-nimg="1"
                              decoding="async"
                              height="40"
                              loading="lazy"
                              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                              style="color: transparent;"
                              width="40"
                            />
                          </p>
                        </button>
                      </a>
                    </div>
                  </span>
                </section>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
              />
              <p
                class="body-1 body-1 ml-11"
                style="display: block; max-width: 100%;"
              >
                Take a free personal health assessment, track your diet and exercise, sync your fitness apps to earn wellness points and moreâ€”all in one secure place.
              </p>
            </div>
          </div>
        </section>
        <section>
          <div
            class="card-main large-section skip-underscore"
          >
            <div
              class="flex flex-col"
            >
              <div
                class="flex flex-row"
              >
                <img
                  alt=""
                  class="size-10"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <section
                  class="flex justify-start self-start my-health-link"
                >
                  <span
                    class="rich-text body-1 undefined"
                  >
                    <div
                      class="flex flex-row body-1 flex-grow md:!flex !block align-top mt-4 ml-2"
                    >
                      <a
                        aria-label="Member Discount"
                        href=""
                        style="display: block;"
                      >
                        <button
                          class="flex flex-row link-container link hover:!underline caremark !flex pt-0 pl-0"
                          style="max-width: max-content; height: auto; display: block;"
                          tabindex="-1"
                          type="button"
                        >
                          <p
                            class="link undefined"
                          >
                            Member Discount
                          </p>
                          <p
                            class="ml-1"
                          >
                            <img
                              alt=""
                              data-nimg="1"
                              decoding="async"
                              height="40"
                              loading="lazy"
                              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                              style="color: transparent;"
                              width="40"
                            />
                          </p>
                        </button>
                      </a>
                    </div>
                  </span>
                </section>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
              />
              <p
                class="body-1 body-1 ml-11"
                style="display: block; max-width: 100%;"
              >
                Want access to new healthy living discounts every week? Find savings on nutrition programs, fitness accessories, medical supplies and services like hearing aids and LASIK eye surgery.
              </p>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</body>
`;
